src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -1062,37 +1074,38 @@ public boolean putOperation(Operation operation, ExecutionStage.Value stage) thr
     return true;
   }
 
-  private void queue(
-      UnifiedJedis jedis,
-      String operationName,
-      List<Platform.Property> provisions,
-      String queueEntryJson,
-      int priority) {
+  private boolean queue(UnifiedJedis jedis, QueueEntry queueEntry, Operation operation)
+      throws IOException {
+    String operationName = operation.getName();
     if (state.dispatchedOperations.remove(jedis, operationName)) {
       log.log(Level.WARNING, format("removed dispatched operation %s", operationName));
     }
-    state.operationQueue.push(jedis, provisions, queueEntryJson, priority);
+    String queueEntryJson = JsonFormat.printer().print(queueEntry);
+    List<Platform.Property> provisions = queueEntry.getPlatform().getPropertiesList();
+    int priority = queueEntry.getExecuteEntry().getExecutionPolicy().getPriority();
+    if (!state.operationQueue.push(jedis, provisions, queueEntryJson, priority)) {
```

Comment:
Stage this with opposite logic fallout: if (...push(...)) return true; if (fallback...push(...)) return true; return false; (with line separations and some comments)